superset/errors.py

```diff
@@ -173,6 +176,12 @@ class SupersetErrorType(str, Enum):
             ),
         },
     ],
+    SupersetErrorType.TEST_CONNECTION_DATABASE_PERMISSIONS_ERROR: [
+        {
+            "code": 1014,
```

Comment:
I think this is code 1017? 
```suggestion
            "code": 1017,
```
https://github.com/apache/superset/pull/14172/files